src/System.Runtime.Extensions/src/System/IO/StreamReader.cs

```diff
@@ -1051,7 +1104,7 @@ public override Task<int> ReadAsync(char[] buffer, int index, int count)
                                 {
                                     if (readToUserBuffer)
                                     {
-                                        n = _decoder.GetChars(tmpByteBuffer, 0, _byteLen, buffer, index + charsRead);
+                                        n = _decoder.GetChars(new ReadOnlySpan<byte>(tmpByteBuffer, 0, _byteLen), buffer.Slice(charsRead).Span, flush: false);
```

Comment:
Is it possible to get the Span from buffer first (outside the outer loop) and then keep slicing on the span since Span.Slice is faster than Memory.Slice?

Reply:
No, this is an async method.